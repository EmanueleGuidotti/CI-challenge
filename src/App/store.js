import { configureStore } from "@reduxjs/toolkit";
import { moviesApi } from "../services/movies.services";
import { setupListeners } from "@reduxjs/toolkit/query";

export const store = configureStore({
  reducer: {
    [moviesApi.reducerPath]: moviesApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }).concat(moviesApi.middleware),
});

// optional, but required for refetchOnFocus/refetchOnReconnect behaviors
// see `setupListeners` docs - takes an optional callback as the 2nd arg for customization
setupListeners(store.dispatch);
