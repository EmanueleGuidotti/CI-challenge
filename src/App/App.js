import { Provider } from "react-redux";
import { store } from "./store";
import { HomepageContainer } from "../pages/Homepage";

const App = () => {
  return (
    <Provider store={store}>
      <HomepageContainer />
    </Provider>
  );
};

export { App };
