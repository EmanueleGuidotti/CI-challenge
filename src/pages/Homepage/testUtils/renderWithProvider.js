import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../../App/store";

const renderWithProviders = (
  ui,
  { preloadedState = {}, myStore = store, ...renderOptions } = {}
) => {
  const Wrapper = ({ children }) => {
    return <Provider store={myStore}>{children}</Provider>;
  };

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
