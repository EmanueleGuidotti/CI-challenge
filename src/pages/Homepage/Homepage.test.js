import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { HomepageContainer } from "./";
import { store } from "../../App/store";
import { renderWithProviders } from "./testUtils/renderWithProvider";
import { mockApiResponse } from "./testUtils/mockApiResponse";

const handlers = [
  rest.get("https://api.themoviedb.org/3/movie/popular", (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockApiResponse));
  }),
];
const server = setupServer(...handlers);

let responseBody;
beforeEach(() => {
  server.use(
    rest.get("https://api.themoviedb.org/3/movie/popular", (req, res, ctx) => {
      responseBody = mockApiResponse;
      return res(ctx.status(200), ctx.json(responseBody));
    })
  );
});

test("Render Homepage loading status properly", async () => {
  const component = renderer.create(
    <Provider store={store}>
      <HomepageContainer />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render Homepage with most viewed movies properly", async () => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
  server.listen();
  const component = renderWithProviders(<HomepageContainer />, store);
  await waitFor(() => {
    expect(responseBody).toEqual(mockApiResponse);
  });
  expect(component).toMatchSnapshot();
});
