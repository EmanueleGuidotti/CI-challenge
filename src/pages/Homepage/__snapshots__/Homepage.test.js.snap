// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Homepage loading status properly 1`] = `
<div
  className="homePageContainer"
>
  <h1>
    My Movies App
  </h1>
  <input
    className="ant-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    style={
      Object {
        "height": 0,
        "overflow": "visible",
        "width": 0,
      }
    }
  />
</div>
`;

exports[`Render Homepage with most viewed movies properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homePageContainer"
      >
        <h1>
          My Movies App
        </h1>
        <input
          class="ant-input"
          placeholder="Search..."
          type="text"
          value=""
        />
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        />
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homePageContainer"
    >
      <h1>
        My Movies App
      </h1>
      <input
        class="ant-input"
        placeholder="Search..."
        type="text"
        value=""
      />
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      />
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
